frontend/src/static/js/v_summary.js

```diff
@@ -145,6 +150,42 @@ window.vSummary = {
                 + `since=${slice.date}'T'00:00:00+08:00&`
                 + `until=${untilDate}'T'23:59:59+08:00`;
     },
+    getFileFormatContributionBars(fileFormatContribution) {
```

Comment:
this function is a bit long. do add some comments or increase the abstraction.

Reply:
I've added in some comments, do let me know if it's ok.

Reply:
Yup, its better.

Reply:
@eugenepeh There's still an error in Codacy for this: ```Arrow function has a complexity of 5. (complexity)```. Can i increase the complexity? It's difficult to abstract the method because there are quite a few common variables used throughout.